src/main/java/com/salesforce/storm/spout/sideline/recipes/trigger/zookeeper/ZookeeperWatchTrigger.java

```diff
@@ -293,18 +285,17 @@ private SidelineRequest buildSidelineRequest(final TriggerEvent triggerEvent) {
 
     /**
      * Using a filter chain step, make some JSON out of it and then hash it to create an idempotent identifier.
+     *
      * @param triggerEvent Trigger event that contains metadata about the request
-     * @param step The FilterChainStep that we are going to generate the request from
      * @return A sideline request identifier for the filter chain step
      * @throws NoSuchAlgorithmException Your java install is whack yo, it's missing MD5, for realz???
      */
     private SidelineRequestIdentifier generateSidelineRequestIdentifier(
-        final TriggerEvent triggerEvent,
-        final FilterChainStep step
+        final TriggerEvent triggerEvent
     ) {
-        final String json = gson.toJson(step);
+        final String data = json.to(triggerEvent.getFilterChainStep());
```

Comment:
it seems like the `TriggerEvent` should perhaps know how to make its identifier.

Reply:
Should be resolved now, let me know what you think @kevpeek